rust/connlib/clients/android/src/lib.rs

```diff
@@ -467,13 +471,14 @@ pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_di
 /// <https://github.com/firezone/firezone/issues/4350>
 ///
 /// # Safety
-/// Pointers must be valid
+/// session_ptr should have been obtained from `connect` function, and shouldn't be dropped with disconnect
+/// at any point before or during operation of this function.
 #[allow(non_snake_case)]
 #[no_mangle]
 pub unsafe extern "system" fn Java_dev_firezone_android_tunnel_ConnlibSession_setDns(
     mut env: JNIEnv,
     _: JClass,
-    session: *const SessionWrapper,
+    session_ptr: jlong,
```

Comment:
What if we changed the order of these parameters to have the pointer be the last one? I guess that could also cause weird behaviour.

Reply:
That's how I figured out this was the cause -- that changed the backtrace of the crash!